File: 1480. Running Sum of 1d Array/main.cpp
#include <vector>

class Solution {
public:
    vector<int> runningSum(vector<int> &nums) {
        vector<int> result;
        int sum = 0;
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];
            result.push_back(sum);
        }

        return result;
    }
};


class Solution {
public:
    vector<int> runningSum(vector<int> &nums) {
        int sum = 0;
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];
            nums[i] = sum;
        }

        return nums;
    }
};

class Solution {
public:
    vector<int> runningSum(vector<int> &nums) {
        for (int i = 1; i < nums.size(); i++) {
            nums[i] += nums[i - 1];
        }

        return nums;
    }
};

File: 1480. Running Sum of 1d Array/main.rs
impl Solution {
    pub fn running_sum(nums: Vec<i32>) -> Vec<i32> {
        let mut sum: i32 = 0;
        nums.iter()
            .map(|x| {
                sum += x;
                sum
            })
            .collect()
    }
}

impl Solution {
    pub fn running_sum(mut nums: Vec<i32>) -> Vec<i32> {
        for i in 1..nums.len() {
            nums[i] += nums[i - 1];
        }

        nums
    }
}
